dhall/src/Dhall/Pretty/Internal.hs

```diff
@@ -420,10 +420,12 @@ prettyCharacterSet characterSet expression =
                 ]
         docsShort (Note  _    c) = docsShort c
         docsShort             c  = [ prettyExpression c ]
-    prettyExpression (Let as b) =
+    prettyExpression (Let x mA a b0) =
         enclose' "" "" space Pretty.hardline
             (fmap duplicate (fmap docA (toList as)) ++ [ docB ])
       where
+        MultiLet as b = multiLet x mA a b0
```

Comment:
This means that pretty-printing depends on whether an expression in de-`Note`d or not: As long as we have the original `Note`s, we can recover the original separation of let-blocks.

I feel this is a bit hacky, but also neat – if we want to keep it, it should at least be documented.

Thoughts?

Reply:
I think that's fine